huaweicloud/provider.go

```diff
@@ -711,6 +711,21 @@ func configureProvider(d *schema.ResourceData, terraformVersion string) (interfa
 		delegated_project = region
 	}
 
+	// Use cloud as basis for identityEndpoint if provided
+	// else use set default endpoint
+	if v, ok := d.GetOk("cloud"); ok && v.(string) != "" {
```

Comment:
The second judgment condition is redundant as GetOk returns true only when the key has been set to a non-zero value.

Reply:
updated.